### PR TITLE
feat: add external secret support

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds external secret support.

This is useful to let the end user bring his own configuration from an existing secret.

### CODE REVIEW

File: charts/dex/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 type: application
 name: dex
-version: 0.2.0
+version: 0.2.1
```
Comment (sagikazarmark) [inline charts/dex/Chart.yaml:4]: Since this is a new feature, please bump the minor versiom.

File: charts/dex/templates/secret.yaml [hunk 2]
```diff
@@ -7,3 +8,4 @@ metadata:
 type: Opaque
 data:
   config.yaml: {{ .Values.config | toYaml | b64enc | quote }}
+{{ end }}
```
Comment (sagikazarmark) [inline charts/dex/templates/secret.yaml:11]: Please adhere .editorconfig settings.

File: charts/dex/values.yaml [hunk 3]
```diff
@@ -33,6 +33,14 @@ grpc:
   # Read more in the [documentation](https://dexidp.io/docs/api/).
   enabled: false
 
+useExternaleSecret:
+  # -- Enable use an external secret for the configuration.
+  enabled: false
+
+  # -- External secret name containing the configuration.
+  # The secret must contain the config.yaml key.
+  name: ''
```
Comment (sagikazarmark) [inline charts/dex/values.yaml:42]: I think I'd prefer something like the `serviceAccount` here:

```suggestion
configSecret:
  # -- Enable creating a secret from the values passed to `config`.
  # If set to false, name must point to an existing secret.
  create: true

  # -- The name of the secret to mount as configuration in the pod.
  # If not set and create is true, a name is generated using the fullname template.
  # Must point to secret that contains at least a `config.yaml` key.
  name: ""
```

File: charts/dex/templates/deployment.yaml [hunk 4]
```diff
@@ -96,7 +96,7 @@ spec:
       volumes:
         - name: config
           secret:
-            secretName: {{ include "dex.fullname" . }}
+            secretName: {{ if .Values.useExternalSecret.enabled }}{{ .Values.useExternalSecret.name }}{{ else }}{{ include "dex.fullname" . }}{{ end }}
```
Comment (sagikazarmark) [inline charts/dex/templates/deployment.yaml:99]: Can you please add a template to the end of the helpers file similar to the service account name template, based on the comments in the values file?

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Thanks @edddycharly for the contribution! I had a few comments, can you please take a look at them?

**Comment by eddycharly** [comment]
@sagikazarmark thanks for your feedback !
I addressed your comments, not sure why the CI was red though.

**Comment by eddycharly** [comment]
Ah, i think I have to use the template in the secret name too

**Comment by eddycharly** [comment]
The condition for creating the secret was wrong, fixed.

**Review by sagikazarmark — APPROVED** [review]

**Comment by sagikazarmark** [comment]
@eddycharly yeah, I think adding a test for that would make sense. You can add a `ci/config-secret-values.yaml` file with the following content:

```yaml
configSecret:
  name: my-super-special-dex-secret
```

That should ensure that the custom template is used everywhere and prevent us from introducing a regression.

**Comment by eddycharly** [comment]
@sagikazarmark I added the ci test file.

**Comment by sagikazarmark** [comment]
Thanks @eddycharly!

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**